tools/close_by_signature.py

```diff
@@ -0,0 +1,243 @@
+                continue
+
+            issue_data.root_issue = msgs_to_root_issue.get(msg)
+            yield issue_data
+            break
+
+        if not msgs_to_search_for:
+            yield issue_data
+
+
+def close_and_link_issues(jira, issues_data_generator):
+    for issue_data in issues_data_generator:
+        if issue_data.root_issue:
+            link_issue_to_root_issue(
+                jira=jira,
+                issue=issue_data.issue,
+                root_issue=issue_data.root_issue,
+            )
+
+        logger.debug('Closing issue with key=%s signature=%s comment=%s',
+                     issue_data.issue.key, type(issue_data.signature),
+                     issue_data.comment.body)
+
+        if dry_run_stdout is None:
+            # set issue status to 'Done'
+            jira.transition_issue(issue_data.issue, '31')
+        else:
+            dry_run_stdout.write(f'Closed issue key={issue_data.issue.key} '
+                                 f'signature={type(issue_data.signature)} '
+                                 f'comment={issue_data.comment.body}\n')
+            dry_run_stdout.flush()
+
+
+def link_issue_to_root_issue(jira, issue, root_issue):
+    logger.info('Linking issue key=%s to root issue key=%s', issue.key,
+                root_issue.key)
+
+    if dry_run_stdout is None:
+        jira.create_issue_link(
+            type='relates to',
+            inwardIssue=issue.key,
+            outwardIssue=root_issue.key,
+        )
+    else:
+        dry_run_stdout.write(f'Linked issue key={issue.key} to root issue '
+                             f'key={root_issue.key}\n')
+
+
+def main(args):
+    config_logger(args.verbose)
+
+    logger.debug('Starting issue resolver task with args=%s',
+                 pprint.pformat(args.__dict__))
+
+    username, password = get_credentials(args.user_password, args.netrc)
+    jira = get_jira_client(username, password)
+
+    signature_types = list(map(
+        lambda x: x.lower().replace('-', '').replace('_', ''),
+        args.signature_type,
+    ))
+    signatures = get_signatures(signature_types, jira)
+
+    issues = get_issues(jira, args.issue, args.recent_issues)
+
+    issues_data_generator = issues_by_signature_generator(
+        jira=jira,
+        signatures=signatures,
+        issues=issues,
+    )
+    issues_data_generator = filter_issues_by_message_generator(
+        jira=jira,
+        issues_data_generator=issues_data_generator,
+        messages=args.message,
+    )
+
+    close_and_link_issues(jira, issues_data_generator)
+
+
+if __name__ == '__main__':
+    logging.basicConfig(level=logging.DEBUG)
+
+    user_args = parse_args()
+
+    if user_args.dry_run_temp:
+        logger.info('Running in dry-run-temp mode')
+        with tempfile.NamedTemporaryFile(mode='w', delete=False) as tmp_fp:
+            dry_run_stdout = tmp_fp
+            main(user_args)
+    else:
+        if user_args.dry_run:
```

Comment:
``elif``

Reply:
> `elif`

Not sure it fits here because `main` gets called outside the internal `if`